contivctl/contivctl.go

```diff
@@ -0,0 +1,287 @@
+package contivctl
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+)
+
+const (
+	DefaultMaster = "http://localhost:9999"
+)
+
+func CreatePolicy(ctx *cli.Context) {
+	argCheck(1, ctx)
+
+	tenant := ctx.String("tenant")
+	policy := ctx.Args()[0]
+
+	logrus.Infof("Creating policy %s:%s", tenant, policy)
+
+	url := fmt.Sprintf("%s%s:%s/", policyURL(ctx), tenant, policy)
+
+	postMap(ctx, url, map[string]interface{}{
+		"tenantName": tenant,
+		"policyName": policy,
+	})
+}
+
+func DeletePolicy(ctx *cli.Context) {
+	argCheck(1, ctx)
+
+	tenant := ctx.String("tenant")
+	policy := ctx.Args()[0]
+
+	logrus.Infof("Deleting policy %s:%s", tenant, policy)
+
+	url := fmt.Sprintf("%s%s:%s/", policyURL(ctx), tenant, policy)
+	deleteURL(ctx, url)
+}
+
+func ListPolicies(ctx *cli.Context) {
+	argCheck(0, ctx)
+
+	tenant := ctx.String("tenant")
+
+	policies := getList(ctx, policyURL(ctx))
+
+	writer := tabwriter.NewWriter(os.Stdout, 0, 2, 2, ' ', 0)
+	defer writer.Flush()
+	writer.Write([]byte("Tenant\tPolicy\n"))
+	writer.Write([]byte("------\t------\n"))
+
+	for _, policy := range policies {
+		if ctx.Bool("all") || policy["tenantName"] == tenant {
+			writer.Write([]byte(fmt.Sprintf("%s\t%s\n", policy["tenantName"].(string), policy["policyName"].(string))))
+		}
+	}
+}
+
+func AddRule(ctx *cli.Context) {
+	argCheck(2, ctx)
+
+	args := map[string]interface{}{
+		"tenantName":    ctx.String("tenant"),
+		"policyName":    ctx.Args()[0],
+		"ruleId":        ctx.Args()[1],
+		"priority":      ctx.Int("priority"),
+		"direction":     ctx.String("direction"),
+		"endpointGroup": ctx.String("epg"),
+		"network":       ctx.String("network"),
+		"ipAddress":     ctx.String("ip-address"),
+		"protocol":      ctx.String("protocol"),
+		"port":          ctx.Int("port"),
+		"action":        ctx.String("action"),
+	}
+
+	if args["ruleId"] == nil || args["ruleId"].(string) == "" {
+		ErrExit(ctx, ExitInvalid, "RuleID (-i) must be specified", false)
+	}
+
+	url := fmt.Sprintf(
+		"%s%s:%s:%s/",
+		ruleURL(ctx),
+		args["tenantName"].(string),
+		args["policyName"].(string),
+		args["ruleId"].(string),
+	)
+
+	postMap(ctx, url, args)
+}
+
+func DeleteRule(ctx *cli.Context) {
+	argCheck(2, ctx)
+
+	tenant := ctx.String("tenant")
+	policy := ctx.Args()[0]
+	ruleId := ctx.Args()[1]
+
+	deleteURL(ctx, fmt.Sprintf("%s%s:%s:%s/", ruleURL(ctx), tenant, policy, ruleId))
+}
+
+func ListRules(ctx *cli.Context) {
+	argCheck(1, ctx)
+
+	tenant := ctx.String("tenant")
+	policy := ctx.Args()[0]
+
+	rules := getList(ctx, ruleURL(ctx))
+
+	writer := tabwriter.NewWriter(os.Stdout, 0, 2, 2, ' ', 0)
+	defer writer.Flush()
+	writer.Write([]byte("Rule\tDirection\tPriority\tEndpointGroup\tNetwork\tIpAddress\tProtocol\tPort\tAction\n"))
```

Comment:
Should we display the output in 'json' like output for example how 'docker inspect' gives the information? I understand that 'docker ps' and 'docker inspect' shows different formats, but 
inspect allows a lot more hierarchical output. I wonder if we need 'inspect' version of a network i.e. epgs and policy details within network, etc.